### PR TITLE
#453: Add new data validation: ListFormulaDataValidation.

### PR DESCRIPTION
**Pull request by civgio**
Fixing the [issue 453:](https://github.com/dhatim/fastexcel/issues/453): a new data validation is added: `ListFormulaDataValidation`.
This `DataValidation` allow to create a data validation with `list` type and with a `formula` value (to referrer a list, for example with `INDIRECT` formula).

In addition, it is possibility to set global visibility to a range when a named range is added.

### CODE REVIEW

**Comment by civgio** [comment]
@meiMingle @rzymek can you check the pull request and if you think the change makes sense, merge it into master?

File: fastexcel-writer/src/main/java/org/dhatim/fastexcel/ListFormulaDataValidation.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+     * Write this dataValidation as an XML element.
+     *
+     * @param w Output writer.
+     * @throws IOException If an I/O error occurs.
+     */
+    @Override
+    public void write(Writer w) throws IOException {
+        w
+                .append("<dataValidation sqref=\"")
+                .append(range.toString())
```
Comment (rzymek) [inline fastexcel-writer/src/main/java/org/dhatim/fastexcel/ListFormulaDataValidation.java:108]: unnessesary .toString()

File: fastexcel-writer/src/main/java/org/dhatim/fastexcel/Range.java [hunk 2]
```diff
@@ -216,13 +233,32 @@ public CustomDataValidation validateWithFormula(String formula) {
...
+    }
+
+    /**
+     * Set the visibility of this range
+     *
+     * @param globalVisibility {@code true} to allow to see the range by all worksheet
+     */
+    public Range setGlobalVisibility(boolean globalVisibility) {
+        this.globalVisibility = globalVisibility;
+        return this;
```
Comment (rzymek) [inline fastexcel-writer/src/main/java/org/dhatim/fastexcel/Range.java:259]: other setters return void. returning this here seems inconsistent

File: fastexcel-writer/src/main/java/org/dhatim/fastexcel/Range.java [hunk 3]
```diff
@@ -44,6 +44,11 @@ public class Range implements Ref {
      */
     private final int right;
 
+    /**
+     * Check about the Range visibility when a named range is added.
```
Comment (rzymek) [inline fastexcel-writer/src/main/java/org/dhatim/fastexcel/Range.java:48]: I find this comment confusing. Doesn't seem to turn on any "Check", but rather control the scope of this range (right?)

**Review by rzymek — COMMENTED** [review]
I would be nice if you'd add some tests covering those changes

File: fastexcel-writer/src/main/java/org/dhatim/fastexcel/ListFormulaDataValidation.java (hunk 1 above)
Comment (civgio) [reply fastexcel-writer/src/main/java/org/dhatim/fastexcel/ListFormulaDataValidation.java:108]: If I try to append the range without "toString", an error is return

File: fastexcel-writer/src/main/java/org/dhatim/fastexcel/Range.java (hunk 2 above)
Comment (civgio) [reply fastexcel-writer/src/main/java/org/dhatim/fastexcel/Range.java:259]: I understand. I change it

**Comment by civgio** [comment]
@rzymek thank you for your comments and your time.
I have fixed all your issues and I have added the tests.
I hope all is ok. If you have any other doubts, I am here.
Have a nice evening.

**Comment by civgio** [comment]
Please @rzymek, can you close the pull request if all is ok? Otherwise can you tell me what needs to be fix to complete the pull request?
Thank you so much for your support and availability

**Review by rzymek — APPROVED** [review]

**Pull request merged**